pkg/operator/ceph/cluster/mon/health.go

```diff
@@ -245,6 +245,51 @@ func (c *Cluster) failMon(monCount, desiredMonCount int, name string) {
 			logger.Errorf("failed to failover mon %q. %v", name, err)
 		}
 	}
+
+	// Check if there are orphaned mon resources that should be cleaned up.
+	// This should only be checked infrequently such as during a mon failover
+	// so we never cleanup mon resources when they are still in use in the middle
+	// of a reconcile.
+	c.removeOrphanMonResources()
+}
+
+func (c *Cluster) removeOrphanMonResources() {
+	logger.Info("checking for orphaned mon resources")
+
+	if c.spec.Mon.VolumeClaimTemplate == nil {
+		logger.Info("skipping check for orphaned mon pvcs")
```

Comment:
```suggestion
		logger.Info("skipping check for orphaned mon pvcs")
```

Did you mean non-pvc?

Reply:
But if the `VolumeClaimTemplate` is nil this means non-pvc right? 🤔

Reply:
That's right. There are no mon pvcs, so we skip checking for them. What about this?
```
logger.Info("skipping check for orphaned mon pvcs since using the host path")
```